src/components/index.js

```diff
@@ -52,7 +52,13 @@ class IProve extends Component {
           }
         })
         requiredSteps.push(step.ast)
-        this.callZ3(requiredSteps, constants, i)
+        this.callZ3(requiredSteps, constants, relations, i)
+      } else {
+       // if there are no dependencies change the box to red
+       if (step.type !== "goal") {
+          // this.callZ3(step.ast, constants, relations, i)
```

Comment:
Remove

Reply:
oh yes sorry that was us trying to fiddle with the boxes when it didnt have any dependencies it should just pass it to z3 that line... but we didnt know how to